src/fsharp/service/service.fs

```diff
@@ -1324,13 +1324,25 @@ type TypeCheckInfo
             let underlyingTy = stripTyEqnsAndMeasureEqns g ty
             isStructTy g underlyingTy
 
+        let isValMutable (vref: ValRef) =
+            // Mutable values, ref cells, byrefs, and outrefs are mutable.
+            // Note that it is not possible for a caller to mutate an inref, so it is not colored, despite being a pointer type.
+            vref.IsMutable
+            || Tastops.isRefCellTy g vref.Type
+            || Tastops.isByrefTy g vref.Type
```

Comment:
Using `isByrefTy` will detect inrefs + outrefs as well, which we don't want to highlight inref values. This check should be this:
```fsharp
vref.IsMutable
|| Tastops.isRefCellTy g vref.Type
|| (Tastops.isByrefTy g vref.Type && not (Tastops.isInByrefTy g vref.Type))
```